model/DataBaseAccess.php

```diff
@@ -65,6 +68,27 @@ protected function getEventManager()
         return $this->getServiceLocator()->get(EventManager::SERVICE_ID);
     }
 
+    public function getResourceTree(core_kernel_classes_Resource $resource): array
+    {
+        $query = <<<'SQL'
+WITH RECURSIVE statements_tree AS (
```

Comment:
note: to check this is compatible with postgres and mysql

Reply:
Checked:
* Postgres
* MariaDB (syntax is similar to MySQL)